### PR TITLE
fix: Cors headers

### PR DESCRIPTION
**Pull request by rkostrzewski**
@developit this change allows only `https://localhost:port` origin to access served app.

### CODE REVIEW

File: src/commands/serve.js [hunk 1]
```diff
@@ -179,11 +179,12 @@ const SERVERS = {
 			options.cwd = persistencePath('preact-cli');
 			process.stderr.write(`Falling back to shared directory + simplehttp2server.\n(dir: ${options.cwd})\n`);
 		}
+		let port = options.port || 8080;
 		return [
 			simplehttp2server,
-			'-cors', '*',
+			'-cors', `https://localhost:${port}`,
```
Comment (developit) [inline src/commands/serve.js]: Can we allow an `options.cors` that gets appended to this as a CSV?

Comment (rkostrzewski) [inline src/commands/serve.js]: Here you go. FYI simplehttp2server is able to add only one origin.

**Comment by rkostrzewski** [comment]
@developit this is a breaking change - as it can break stuff on things like now.sh

**Comment by developit** [comment]
@rkostrzewski maybe we should check if there is an env var we can watch for?

**Comment by rkostrzewski** [comment]
@developit we can check for `NOW` variable on now.sh, but I don't really think it's worth the trouble as far more providers exist out there. Furthermore setting CORS to * even on hosting provider doesn't seem like a good idea.

**Comment by reznord** [comment]
Looks like it passed all the tests. Can I merge this?

**Comment by rkostrzewski** [comment]
I'd wait for @developit since IMHO this is a major version bump and maybe something smaller will come along.

**Comment by developit** [comment]
I think I overestimated the breakage this would cause initially - really the only time someone would run into an issue is if they are trying to fetch assets from their preact-cli output on a page with a differing origin. I think that's a fairly small set of people - I've had such a usecase before, but I would expect to have to muck around with hostnames/headers a bit to make things work.

Should we wait for a bit of breakage to accumulate and make a 2.0.0, or merge and cross our fingers? haha

**Comment by lukeed** [comment]
TBH, I'm not fully aware of the implications, but I have a feeling that (in development) that would be a tiny tiny subset of people. 

In production, I'd assume that the requesting origins would be known & specified... or just reverted back to `"*"`.

**Comment by developit** [comment]
The last time I ran into this was proxying my initial bundle, but leaving it to reference its associated webpack assets from a CDN.  The thing is, I had to manually configure Webpack to set the public path to the CDN.

Maybe we could generate a list of allowed origins using some more information:

- the user's own ip (or maybe `192.*`?)
- any domain found in `publicPath`
- the `host` / `hosts` CLI arg
- `$HOSTNAME`

The combination of these would handle a whole bunch of cases, really leaving only a few where there's not enough information to determine safety.

**Comment by rkostrzewski** [comment]
Simple http2server doesn't support multiple origins :cry:

**Comment by developit** [comment]
(continuing from slack lol) When you suggested setting the header from `firebase.json` - do you know if that works with simplehttp2server?

**Pull request merged**